### ISSUE TITLE
Error: train_batch_size inconsistence

### ISSUE THREAD

**Issue by bing0037** [microsoft/DeepSpeed#1306]
Hi,

I am trying to reproduce GPT2 pretraining using this example ([ds_pretrain_gpt2-zero3.sh](https://github.com/microsoft/DeepSpeedExamples/blob/master/Megatron-LM-v1.1.5-ZeRO3/examples/ds_pretrain_gpt2-zero3.sh)). 

I am confused why we set `BATCHSIZE` in [ds_pretrain_gpt2-zero3.sh](https://github.com/microsoft/DeepSpeedExamples/blob/master/Megatron-LM-v1.1.5-ZeRO3/examples/ds_pretrain_gpt2-zero3.sh) while in the meantime setting `train_batch_size` in ds_zero_stage_3_config.json. They seem in **conflict** and I think it might be a BUG.

I debugged the code and found that DeepSpeed:train_batch_size only affects the calculation of `global_samples` (just used for tensorboard logging) in class DeepSpeedEngine([engine.py](https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/runtime/engine.py)).

Did anyone design it on purpose? It seems a bug since the value of x-axis doesn't match the real value.

## Environment:
```
torch=1.9.0
deepspeed=0.4.4
```
Any thought or explanation? Thanks.

**Comment by tjruwase** [comment]
@bing0037, yes this is an aspect of deepspeed that is prone to subtle errors. We currently don't have a robust solution to preventing mismatches in client and deepspeed views of batch sizes and are open to ideas. Please see more discussions in #84  and #152.

**Comment by tjruwase** [comment]
@bing0037, do you still need this issue opened? Thanks.
